src/formatDate.js

```diff
@@ -50,7 +50,34 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const dateObj = {};
+  const currentYear = new Date().getFullYear().toString();
+
+  for (let i = 0; i < dateArr.length; i++) {
+    dateObj[fromFormat[i]] = dateArr[i];
+  }
+
+  const fromYearFormat = toFormat.find((el) => el.startsWith('Y'));
+  const toYearFormat = fromFormat.find((el) => el.startsWith('Y'));
+
+  if (fromYearFormat.length === 4 && toYearFormat.length === 2) {
+    dateObj[fromYearFormat]
+      = dateObj[toYearFormat] * 1 > Number(currentYear.slice(-2))
+        ? '19' + dateObj[toYearFormat]
+        : '20' + dateObj[toYearFormat];
```

Comment:
19 and 20 are 'magic number', so please move these to the constants

Reply:
I try to did this, but code was less readable, so i did it from scratch.